src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/MigrateBookingStatusTest.kt

```diff
@@ -0,0 +1,321 @@
+        assertBookingStatusIsNull(booking2, ServiceName.temporaryAccommodation)
+
+        migrationJobService.runMigrationJob(MigrationJobType.bookingStatus, 1)
+
+        assertBookingStatus(booking1, BookingStatus.departed, ServiceName.temporaryAccommodation)
+        assertBookingStatus(booking2, BookingStatus.departed, ServiceName.temporaryAccommodation)
+      }
+    }
+  }
+
+  @Test
+  fun `Successfully migrate multiple CAS3 provisional booking and returns 202 response`() {
+    `Given a User` { userEntity, jwt ->
+      `Given an Offender` { offenderDetails, _ ->
+        val booking1 = createTemporaryAccommodationBooking(userEntity, null)
+        val booking2 = createTemporaryAccommodationBooking(userEntity, null)
+
+        assertBookingStatusIsNull(booking1, ServiceName.temporaryAccommodation)
+        assertBookingStatusIsNull(booking2, ServiceName.temporaryAccommodation)
+
+        migrationJobService.runMigrationJob(MigrationJobType.bookingStatus, 1)
+
+        assertBookingStatus(booking1, BookingStatus.provisional, ServiceName.temporaryAccommodation)
+        assertBookingStatus(booking2, BookingStatus.provisional, ServiceName.temporaryAccommodation)
+      }
+    }
+  }
+
+  @Test
+  fun `Successfully migrate multiple CAS3 Arrived booking and returns 202 response`() {
+    `Given a User` { userEntity, jwt ->
+      `Given an Offender` { offenderDetails, _ ->
+        val booking1 = createCAS3ArrivedBooking(userEntity)
+        val booking2 = createCAS3ArrivedBooking(userEntity)
+
+        assertBookingStatusIsNull(booking1, ServiceName.temporaryAccommodation)
+        assertBookingStatusIsNull(booking2, ServiceName.temporaryAccommodation)
+
+        migrationJobService.runMigrationJob(MigrationJobType.bookingStatus, 1)
+
+        assertBookingStatus(booking1, BookingStatus.arrived, ServiceName.temporaryAccommodation)
+        assertBookingStatus(booking2, BookingStatus.arrived, ServiceName.temporaryAccommodation)
+      }
+    }
+  }
+
+  @Test
+  fun `Successfully migrate multiple CAS3 Confirmed booking and returns 202 response`() {
+    `Given a User` { userEntity, jwt ->
+      `Given an Offender` { offenderDetails, _ ->
+        val booking1 = createCAS3ConfirmedBooking(userEntity)
+        val booking2 = createCAS3ConfirmedBooking(userEntity)
+
+        assertBookingStatusIsNull(booking1, ServiceName.temporaryAccommodation)
+        assertBookingStatusIsNull(booking2, ServiceName.temporaryAccommodation)
+
+        migrationJobService.runMigrationJob(MigrationJobType.bookingStatus, 1)
+
+        assertBookingStatus(booking1, BookingStatus.confirmed, ServiceName.temporaryAccommodation)
+        assertBookingStatus(booking2, BookingStatus.confirmed, ServiceName.temporaryAccommodation)
+      }
+    }
+  }
+
+  @Test
+  fun `Successfully migrate CAS3 only 'departed' bookings when request is for 'departed' bookings and multiple booking exists in the DB`() {
```

Comment:
I was hoping to see a test that tried a combination of bookings in various states, woop! 🙌